door-open-service/node-backend/src/slackInterface.js

```diff
@@ -23,6 +23,19 @@ export default function (config, slackBot, sockServer) {
           "Consider dispatching a drone to pick up a human.");
 
     } else {
+
+      config.doorTimes = { "garage": 25 * 1000 };
+
+      if (config.lock) {
+        if (config.lock.user!==user.email && (+new Date() - config.lock.time)< config.doorTimes["garage"]) {
+            //Another user is trying to manage the door, and the door is still within lock time, so we must abort
+          slackBot.postMessage(`Opening the garage denied by a pending lock of ${config.lock.user}. ...`);
+          return;
+        }
+        config.lock.user=user.email, config.lock.time=+new Date(); //update lock data
+      } else {
+        config.lock={ user:user.email, time:+new Date() };
```

Comment:
This lines does the same as the one inside the `if`. You can always update the lock, regardless of whether the lock is set or not.